src/main/network/router/token_bucket.rs

```diff
@@ -0,0 +1,154 @@
+use std::time::Duration;
+
+use crate::core::support::emulated_time::EmulatedTime;
+use crate::core::support::simulation_time::SimulationTime;
+use crate::core::worker::Worker;
+
+pub struct TokenBucket {
+    capacity: u64,
+    balance: u64,
+    refill_size: u64,
+    refill_interval: Duration,
+    last_refill: EmulatedTime,
```

Comment:
OK. I think we want to remove use of `SimulatedTime` in rust as much as possible, based on this comment:

https://github.com/shadow/shadow/blob/26d5c728d7d1edabc48c8b36b8a2128e809fd489/src/main/core/support/simulation_time.rs#L4-L6

But there currently are not many helpers in `EmulatedTime` for converting between `Duration` types.

Reply:
Ah okay. So we probably want to change the methods of `EmulatedTime` to return `Duration`s instead of `SimulationTime`s then right?

Reply:
I think that comment is outdated. I've since found it useful to use SimulatedTime and expanded its functionality a bit.

Reply:
I'm still on the fence whether it's especially useful to have different types for a "simulated duration" vs a "real duration", but it's definitely useful to have a duration-like that we own. e.g. we have implemented `impl std::convert::TryFrom<libc::timeval> for SimulationTime`, but aren't allowed to implement `impl std::convert::TryFrom<libc::timeval> for Duration` since we don't own either type.

Reply:
Actually I think the more compelling reason is that `SimTime` in C is 64-bits, while `Duration` is effectively 128-bits.

Using the 128-bit `Duration` in Rust code would mean moving the check if a value fits in 64 bits to the C/Rust boundaries, instead of the places where we actually get or derive the value, which both makes the code a bit messier, and would make a conversion error a bit harder to debug.

Reply:
OK so I think the guidance here is to remove `Duration` and just use `SimulationTime`, and maybe update the outdated comment.

(Perhaps we should rename `EmulatedTime` and `SimulationTime` to `EmulatedInstant` and `EmulatedDuration` or some such, but I'll leave that bikeshed for later ;)

Reply:
Done, moved away from `Duration` and expanded and used `SimulationTime` instead.